core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -870,4 +871,8 @@ public MoveType type() {
   private Types.NestedField findField(String fieldName) {
     return caseSensitive ? schema.findField(fieldName) : schema.caseInsensitiveFindField(fieldName);
   }
+
+  private String caseSensitiveName(String name) {
```

Comment:
it can return both casesensitive as well as case insensitive name, how about calling it caseSensitivityAwareName() ?